client/src/components/ShelfCreator.js

```diff
@@ -0,0 +1,341 @@
+            return (
+              <SubShelf
+                key={'column' + index}
+                shelfWidth={shelfWidth(index)}
+                shelfHeight={shelfHeight(index)}
+                child={index}
+                getColor={getShelfBorders(shelf.color)}
+                data-testid="sub-shelf"
+              >
+                {column.compartments &&
+                  column.compartments.length > 0 &&
+                  column.compartments.map((compartment, index) => {
+                    return (
+                      <Compartment
+                        key={'compartment' + index}
+                        getColor={getShelfBorders(shelf.color)}
+                        data-testid="compartment"
+                      />
+                    );
+                  })}
+              </SubShelf>
+            );
+          })}
+        </ShelfPreview>
+      </ShelfArea>
+    </>
+  );
+}
+
+const ShelfArea = styled.section`
+  height: 100%;
+  margin: 1rem auto 9rem;
+  width: 95%;
+
+  input,
+  select {
+    border: 1px solid var(--primary);
+    border-radius: var(--border-radius);
+    background: var(--background);
+  }
+`;
+const ShelfStarter = styled.section`
+  display: flex;
+  justify-content: space-around;
+  margin: 1rem auto 2rem;
+  label {
+    display: block;
+    font-size: 0.8rem;
+    margin-bottom: 0.2rem;
+  }
+
+  input,
+  select {
+    box-shadow: var(--box-shadow-offset-x) var(--box-shadow-offset-y)
+      var(--box-shadow-blur) var(--box-shadow-color);
+    height: 2rem;
+  }
+`;
+
+const ShelfConfigWrapper = styled.section`
+  margin: 1rem auto 2rem;
+  width: 90%;
+`;
+
+const ShelfConfigHeader = styled.div`
+  display: flex;
+  gap: 1rem;
+  justify-content: space-around;
+
+  p {
+    font-size: 0.6rem;
+  }
+`;
+
+const ShelfConfig = styled.div`
+  align-items: center;
+  border-radius: var(--border-radius);
+  box-shadow: var(--box-shadow-offset-x) var(--box-shadow-offset-y)
+    var(--box-shadow-blur) var(--box-shadow-color);
+  display: flex;
+  justify-content: space-around;
+  margin: 0.5rem auto;
+  padding: 0.5rem;
+  width: 100%;
+
+  p {
+    font-size: 0.9rem;
+  }
+
+  label {
+    display: block;
+    font-size: 0.7rem;
+    margin-bottom: 0.2rem;
+  }
+
+  :not(:first-child) {
+    label {
+      display: none;
+    }
+  }
+
+  :nth-child(odd) {
+    background: var(--primary);
+
+    select {
+      border-color: var(--secondary);
+    }
+  }
+  :nth-child(even) {
+    background: var(--secondary);
+    color: var(--background);
+  }
+
+  select {
+    height: 1.5rem;
+    width: 3rem;
+  }
+`;
+
+const ShelfPreview = styled.section`
+  align-items: flex-end;
+  display: flex;
+  gap: 1px;
+  justify-content: center;
+  margin: 1rem auto;
+  height: 150px;
+  width: 90%;
+`;
+
+const SubShelf = styled.div`
+  border: ${(props) => props.getColor};
+  display: flex;
+  flex-direction: column;
+  height: ${(props) => props.shelfHeight}%;
```

Comment:
See comment above: `shelfHeight` is a function that should be invoked in order to retrieve a value. Or does `styled-component` perform some magic here? ;)